openscap-policies/src/main/scala/com/normation/plugins/openscappolicies/api/OpenScapApi.scala

```diff
@@ -0,0 +1,133 @@
+    }}.toList
+  }
+
+  def response(function: Box[JValue], req: Req, errorMessage: String, id: Option[String], dataName : String)(implicit action: String): LiftResponse = {
+    RestUtils.response(restExtractorService, dataName, id)(function, req, errorMessage)
+  }
+
+  object GetOpenScapReport extends LiftApiModule {
+    val schema = OpenScapApi.GetOpenScapReport
+    val restExtractor = api.restExtractorService
+
+    def process(version: ApiVersion, path: ApiPath, nodeId: String, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+      implicit val action = "getOpenScapReport"
+      (for {
+        report <- openScapReportReader.getOpenScapReport(NodeId(nodeId))
+      } yield {
+        logger.info(s"Report for node ${nodeId} has been found ")
+        report
+      }) match {
+        case Full(report) =>
+          logger.trace("doing in memory response")
+          InMemoryResponse(
+              report.content.getBytes()
+            , ("Content-Type" -> "text/html") ::
+              ("Content-Disposition","""attachment;filename="rudder-openscap-%s.html"""".format(nodeId)) ::
+              Nil
+            , Nil
+            , 200)
+        case eb: EmptyBox =>
+          val errorMessage = eb ?~! "Could not get the OpenScap report for node ${nodeId}"
+          logger.error(errorMessage.messageChain)
+          val html = <div class="error">{errorMessage.messageChain}</div>
+          InMemoryResponse(
+              errorMessage.messageChain.getBytes()
+            , Nil
+            , Nil
+            , 404)
+      }
+
+    }
+  }
+
+  object GetSanitizedOpenScapReport extends LiftApiModule {
+    val schema = OpenScapApi.GetSanitizedOpenScapReport
+    val restExtractor = api.restExtractorService
+
+    def process(version: ApiVersion, path: ApiPath, nodeId: String, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+      implicit val action = "getSanitizedOpenScapReport"
+      (for {
+        report        <- openScapReportReader.getOpenScapReport(NodeId(nodeId)) ?~! s"Cannot get OpenScap Report for node ${nodeId}"
+        sanitizedXml  <- reportSanitizer.sanitizeReport(report).toBox ?~! "Error while sanitizing report"
+      } yield {
+        logger.trace(s"Report for node ${nodeId} has been found and sanitized")
+        sanitizedXml
+      }) match {
+        case Full(sanitizedReport) =>
+          logger.trace("Doing in memory response")
+          InMemoryResponse(
+            sanitizedReport.toString().getBytes()
+            // somehow, the X-FRAME-OPTIONS get rewrote to DENY here
+            ,  ("Content-Type", "text/html") :: ("Content-Disposition", "inline") :: Nil
+            , Nil
+            , 200)
+        case eb: EmptyBox =>
+          val errorMessage = eb ?~! "Could not get the sanitized OpenScap report for node ${nodeId}"
+          logger.error(errorMessage.messageChain)
```

Comment:
hum, in fact it won't work in your case, because that API call is use only internally to display the result directly. 
So that API should really be only internal I think